### ISSUE TITLE
Options from extended tsconfig.json are ignored

### ISSUE THREAD

**Issue by wangweixuan** [ezolenko/rollup-plugin-typescript2#131]
#### What happens and why it is wrong
When parsing tsconfig.json, the options are overridden using a naive merge method before they are actually parsed. This leads to certain options (e.g. `module`) extended from a base tsconfig being ignored unexpectedly and unconditionally.
https://github.com/ezolenko/rollup-plugin-typescript2/blob/6b0225a7de935780ce75a811d116deee428b3313/src/parse-tsconfig.ts#L45-L50

#### Versions
All latest.

#### tsconfig.json
```jsonc
{
  "extends": "./tsconfig.base"
}
```

#### tsconfig.*base*.json
```jsonc
{
  "compilerOptions": {
    "module": "esnext"
  }
}
```

#### plugin output with verbosity 3
```jsonc
// truncated
rpt2: parsed tsconfig: {
    "options": {
        "module": 5,                                // UNEXPECTED
        "configFilePath": "/path/to/tsconfig.json",
        // truncated
    },
    // truncated
    "raw": {
        "compilerOptions": {
            "module": "ES2015"                      // UNEXPECTED
        },
        "extends": "./tsconfig.base",
        "compileOnSave": false
    },
    // truncated
}
// truncated
```

**Comment by ezolenko** [comment]
Yep, only affects module. Should be fixed in master now

**Comment by ezolenko** [comment]
Fixed in 0.19.2
